elasticdl/python/tests/worker_pytorch_training_loop_test.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2020 The ElasticDL Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import unittest
+
+import tensorflow as tf
+import torch
+
+from elasticdl.proto import elasticdl_pb2
+from elasticdl.python.common.args import parse_worker_args
+from elasticdl.python.tests.test_utils import (
+    create_pserver,
+    get_mnist_dataset,
+    get_random_batch,
+)
+from elasticdl.python.worker.ps_client import PSClient
+from elasticdl.python.worker.worker_pytorch import WorkerPytorch
+from elasticdl_client.common.constants import DistributionStrategy
+
+sys.path.append("../")
+
+
+class WorkerPSInteractionTest(unittest.TestCase):
+    def setUp(self):
+        self._model_zoo_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), "../../../model_zoo"
+        )
+        self._batch_size = 16
+        self._channels = []
+        self._pservers = []
+        self._workers = []
+
+    def tearDown(self):
+        for pserver in self._pservers:
+            pserver.server.stop(0)
+
+    def _create_pserver(self, model_def, num):
+        self._ports, self._channels, self._pservers = create_pserver(
+            self._model_zoo_path,
+            model_def,
+            grads_to_wait=1,
+            use_async=True,
+            num_ps_pods=num,
+        )
+        self._model_def = model_def
+
+    def _create_worker(self, worker_num):
```

Comment:
The function  is  the same as `create_worker` in 
https://github.com/sql-machine-learning/elasticdl/blob/5fe8f4bc01cb0358926066b8d9dd014692e05f26/elasticdl/python/tests/worker_ps_interaction_test.py#L65-L85

You can move both them into `test_utils.py` and import it.

Reply:
Thanks. Good suggestion.